tools/config_parser.c

```diff
@@ -238,9 +239,13 @@ static int __mmap_parse_file(const char *path, process_entry_fn *process_entry)
 		goto out_unref;
 	}
 
-	for (len = g_mapped_file_get_length(file);
-	     len > 0 && len != (size_t)-1;
-	     len -= delim - contents + 1, contents = delim + 1) {
+	len = g_mapped_file_get_length(file);
+	if (len == 0 || len == SIZE_MAX) {
+		ret = -EINVAL;
+		goto out_close;
```

Comment:
g_mapped_file_unref() is not called on error handling ?. So, should we use goto out_unref instead of out_close ?